### ISSUE TITLE
"Not a keyfile??"

### ISSUE THREAD

**Issue by rwdean** [HardCorePawn/multibit_recovery#9]
Trying to open my encrypted keys, but from the command window i get a 'not a keyfile??' notification. Same error hands with newly created, empty encrypted keys as well.

**Comment by HardCorePawn** [comment]
which version of multibit are you running? and can you provide one of the newly created, empty encrypted .key files (and password) as well so that I can inspect/test it?

**Comment by rwdean** [comment]
Multibit Classic Version 0.5.19 ok, i spent some more time on it. Running -python decrypt_multibit_classic_keys.py multibit-\<DATETIMESTAMP>.key- from the command line gives this error every time, but dragging the key into the script works. However the prompt there does not display that text is typing, so it was a bit confusing but worked once i realized it was accepting input.

It seems my key has been corrupted, as it finally gave me a key, but its all wingdings.

**Comment by HardCorePawn** [comment]
Just out of curiousity... if you open the .key file in a text editor... does it begin with the text "U2F" or something else? The "not a .key file" error occurs if the file does not appear to be a base64 encoded file starting with a specific byte sequence.

Was this file generated on a computer set with English as the OS language? or was it using something else? It's possible the file encoding might be causing issues.

Also, did your password contain "non-english" alphabet characters? ie. accented characters from European languages? ???

**Comment by rwdean** [comment]
Yes, it does begin with U2F, it was generated on an English US computer, and all password characters are standard, non-accented letters and/or numbers.

**Comment by HardCorePawn** [comment]
And what happens if you create a new empty wallet in MultiBit Classic... and export the .key file... does it give "wingdings" or does it decrypt successfully?

If a new .key file is OK... but the old one is spitting out garbage... then it's possible the original .key file is corrupt :(

**Comment by rwdean** [comment]
Yeah that's what's happening. New ones are readable keys. it's gotta be corrupt.

**Comment by 0xdevalias** [comment]
Not sure if this will be helpful to anyone here, but I was initially getting the `Not a keyfile??` error because I was trying to run the script with python 3.x; which isn't supported. Switching back to python `2.x` resolved it for me:

- https://github.com/HardCorePawn/multibit_recovery/issues/10
  - > These scripts only work with Python 2

Not sure if this will be helpful to anyone here, but I found that for an encrypted key that starts with `U2F` and can be decrypted correctly like this:

```shell
⇒ python decrypt_multibit_classic_keys.py example.key
```

I was also able to decrypt it using openssl directly like this:

```shell
⇒ openssl enc -d -aes-256-cbc -md md5 -base64 -in example.key -out example-decrypted.key
enter AES-256-CBC decryption password:
*** WARNING : deprecated key derivation used.
Using -iter or -pbkdf2 would be better.
```

And when I was using the wrong password, it would look more like this:

```shell
⇒ openssl enc -d -aes-256-cbc -md md5 -base64 -in example.key -out example-decrypted.key
enter AES-256-CBC decryption password:
*** WARNING : deprecated key derivation used.
Using -iter or -pbkdf2 would be better.
bad decrypt
00D33056F87F0000:error:1C800064:Provider routines:ossl_cipher_unpadblock:bad decrypt:providers/implementations/ciphers/ciphercommon_block.c:112:
```
